api/src/main/java/org/apache/iceberg/catalog/SupportsNamespaces.java

```diff
@@ -0,0 +1,140 @@
+  }
+
+  /**
+   * Create a namespace in the catalog.
+   *
+   * @param namespace a multi-part namespace
+   * @param metadata a string Map of properties for the given namespace
+   * @throws AlreadyExistsException If the namespace already exists
+   * @throws UnsupportedOperationException If create is not a supported operation
+   */
+  void createNamespace(Namespace namespace, Map<String, String> metadata);
+
+  /**
+   * List top-level namespaces from the catalog.
+   * <p>
+   * If an object such as a table, view, or function exists, its parent namespaces must also exist
+   * and must be returned by this discovery method. For example, if table a.b.t exists, this method
+   * must return ["a"] in the result array.
+   *
+   * @return an List of namespace {@link Namespace} names
+   */
+  default List<Namespace> listNamespaces() {
+    return listNamespaces(Namespace.empty());
+  }
+
+  /**
+   * List namespaces from the namespace.
+   * <p>
+   * For example, if table a.b.t exists, use 'SELECT NAMESPACE IN a' this method
+   * must return Namepace.of("a","b") {@link Namespace}.
+   *
+   * @return a List of namespace {@link Namespace} names
+   * @throws NoSuchNamespaceException If the namespace does not exist (optional)
+   */
+  List<Namespace> listNamespaces(Namespace namespace) throws NoSuchNamespaceException;
+
+  /**
+   * Load metadata properties for a namespace.
+   *
+   * @param namespace a namespace. {@link Namespace}
+   * @return a string map of properties for the given namespace
+   * @throws NoSuchNamespaceException If the namespace does not exist (optional)
+   */
+  Map<String, String> loadNamespaceMetadata(Namespace namespace) throws NoSuchNamespaceException;
+
+  /**
+   * Drop namespace, while the namespace haven't table or sub namespace will return true.
+   *
+   * @param namespace a namespace. {@link Namespace}
+   * @return true if the namespace was dropped, false otherwise.
+   */
+  boolean dropNamespace(Namespace namespace);
+
+  /**
+   * Apply a set of metadata to a namespace in the catalog.
+   *
+   * @param namespace a namespace. {@link Namespace}
+   * @param properties a collection of metadata to apply to the namespace
+   * @throws NoSuchNamespaceException If the namespace does not exist (optional)
+   * @throws UnsupportedOperationException If namespace properties are not supported
+   */
+  boolean setProperties(Namespace namespace, Map<String, String> properties) throws NoSuchNamespaceException;
+
+  /**
+   * Remove a set of metadata from a namespace in the catalog.
+   *
+   * @param namespace a namespace. {@link Namespace}
+   * @param properties a collection of metadata to apply to the namespace
+   * @throws NoSuchNamespaceException If the namespace does not exist (optional)
+   * @throws UnsupportedOperationException If namespace properties are not supported
+   */
+  boolean removeProperties(Namespace namespace, Set<String> properties) throws NoSuchNamespaceException;
+
```

Comment:
Please remove extra newlines like this one.

Reply:
OK